### PR TITLE
Add ParagraphDecoration

### LINKED ISSUE #126: Paragraph decorations
We need API for paragraph alignment, spacing, top/bottom indentation

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #126 

By adding a `ParagraphDecoration` to `Piece`, the model can track changes in the text (`TextDecoration`/`ImageDecoration`) but also in the paragraphs (defined between line feeds) which contain those pieces, and actions work as expected for any kind of decoration, including undo/redo support.

### CODE REVIEW

File: src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java [hunk 1]
```diff
@@ -76,8 +88,9 @@ protected void invalidated() {
             if (!hasSelection()) {
                 Decoration decorationAtCaret = getTextBuffer().getDecorationAtCaret(get());
                 if (decorationAtCaret instanceof TextDecoration) {
-                    setDecoration(decorationAtCaret);
+                    setDecorationAtCaret(decorationAtCaret);
                 }
+                getParagraphWithCaret().ifPresent(p -> setDecorationAtParagraph(p.getDecoration()));
```
Comment (abhinayagarwal) [inline src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java:93]: Since TextDecoration and ParagraphDecoration are mutually exclusive, the order doesn't matter. Is my understanding correct?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java:93]: yes, indeed, you can't apply both at the same time, so order doesn't matter.

Comment (abhinayagarwal) [reply src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java:93]: Preset paragraph style (#76) is currently implemented (#124) using TextDecorations. Do you see the implementation getting changed because of this PR?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java:93]: Paragraph styles should be actually a combination of TextDecoration and ParagraphDecoration, I don't see a problem with that so far.

File: src/main/java/com/gluonhq/richtext/action/TextDecorateAction.java [hunk 2]
```diff
@@ -30,8 +30,8 @@ protected void bind() {
             if (nv != null && !updating) {
                 updating = true;
                 TextDecoration.Builder builder = TextDecoration.builder();
-                Decoration decoration = viewModel.getDecoration();
-                if (viewModel.getSelection() == Selection.UNDEFINED && decoration instanceof TextDecoration) {
+                Decoration decoration = viewModel.getDecorationAtCaret();
+                if (viewModel.getSelection().isDefined() && decoration instanceof TextDecoration) {
```
Comment (abhinayagarwal) [inline src/main/java/com/gluonhq/richtext/action/TextDecorateAction.java:34]: I am trying to understand why this change is done here?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/richtext/action/TextDecorateAction.java:34]: Hmm, I might have missed an `!`?

**Review by abhinayagarwal — APPROVED** [review]
